compiler/rustc_symbol_mangling/src/legacy.rs

```diff
@@ -239,6 +249,13 @@ impl<'tcx> Printer<'tcx> for SymbolPrinter<'tcx> {
                 self.write_str("[")?;
                 self.print_type(ty)?;
                 self.write_str("; ")?;
+                let size = self
```

Comment:
Existing but I think this should just be calling `self.print_const`, if `print_const` isn't handling everything that this code is then it ought to be updated so that the "better" printing of const args to arrays is also present for *all* const arguments.

It seems somewhat error prone to be implementing mangling of consts in two separate places

Reply:
Yea, I didn't want to dedup that initially because I couldn't tell if it was somehow load bearing. I'll do that now

Reply:
hmm... the reason this was done was to avoid printing `0_usize` for array lengths... I don't see a good way to avoid doing that if we want `print_const` to normally print the type, too.

Reply:
Ah unfortunate